From 8e98411c927d37603ad92710928f683055dc40ba Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 22 Jan 2021 00:26:44 +0200
Subject: [PATCH] [SW-0] gaudi: reset: reset uart/spi clk to ref clk

Since we raise the PCI PLL in Preboot now, upon reset we get the BTL and
early Ppboot messages gibberished. This is happening because BTL expects
the UART/SPI clock to be at the reference clock frequency.
Reset the UART/SPI clock to reference clock before the actual reset
fires.

Change-Id: I212af52fbb010854020beb9d91e7d973ec454f4e
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/reset.c b/gaudi/reset.c
index f0e54a6..86d680c 100644
--- a/gaudi/reset.c
+++ b/gaudi/reset.c
@@ -6,6 +6,7 @@
 
 #include <gaudi_blocks.h>
 #include <psoc_global_conf_regs.h>
+#include <psoc_pci_pll_regs.h>
 #include <pcie_wrap_regs.h>
 #include <pcie_core_regs.h>
 #include <pcie_dbi_regs.h>
@@ -376,6 +377,9 @@
 	/* Restart BTL/BLR upon hard-reset */
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_BOOT_SEQ_RE_START);
 
+	/* move the UART/SPI clock to reference clock */
+	regmap_writel(HL_PLL_REF_CLK, mmPSOC_PCI_PLL_DIV_SEL_2);
+
 	udelay(1000);
 
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_SW_ALL_RST);
